Use aligned_alloc where possible

LumioseSil opened the pull request:
#### 42ce1889929fe99b82b78440120ec6f4a4ca57f8
<pre>
Use aligned_alloc where possible
<a href="https://bugs.webkit.org/show_bug.cgi?id=266538">https://bugs.webkit.org/show_bug.cgi?id=266538</a>

Reviewed by Michael Catanzaro.

aligned_alloc is more portable and cleaner than
posix_memalign is. In fact, aligned_alloc
supersedes posix_memalign entirely.

* Source/WTF/wtf/FastMalloc.cpp: Ditto.
  (WTF::fastAlignedMalloc): Replace posix_memalign
  with aligned_alloc.
  (WTF::tryFastAlignedMalloc): Ditto.
* Source/bmalloc/bmalloc/DebugHeap.cpp: Ditto.
  (bmalloc::DebugHeap::memalign): Ditto.

Canonical link: <a href="https://commits.webkit.org/279189@main">https://commits.webkit.org/279189@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9e45716157f5f53e9b0ab171daf10d0b1969b4d1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52751 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32088 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5235 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56029 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3474 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38748 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3196 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42821 "Found 1 new test failure: http/tests/scroll-to-text-fragment/no-scroll-after-stylesheet-load.html (failure)") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2235 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54849 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45531 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23927 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/26898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2840 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1633 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/46104 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/48777 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2993 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57621 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/52264 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27888 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2989 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50221 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29112 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45723 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49496 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11526 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30027 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/64568 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28864 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12252 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on PerformanceTests/MallocBench/MallocBench/mbmalloc.cpp:
```diff
@@ -42,10 +42,7 @@ void* mbmalloc(size_t size)
-    if (posix_memalign(&result, alignment, size))
-        return nullptr;
-    return result;
+    return aligned_alloc(alignment, size);
```
I wouldn't touch MallocBench at all because it's a benchmark. I bet this _probably_ won't affect performance, but I don't know for sure, and we don't want to be wrong. It's OK for MallocBench to follow older patterns than the rest of WebKit.

mcatanzaro reviewed (commented):
I agree `aligned_alloc` is better. Nice change. One concern, though:

LumioseSil committed:
Use aligned_alloc where possible
https://bugs.webkit.org/show_bug.cgi?id=266538

Reviewed by Michael Catanzaro.

aligned_alloc is more portable and cleaner than
posix_memalign is. In fact, aligned_alloc
supersedes posix_memalign entirely.

* Source/WTF/wtf/FastMalloc.cpp: Ditto.
  (WTF::fastAlignedMalloc): Replace posix_memalign
  with aligned_alloc.
  (WTF::tryFastAlignedMalloc): Ditto.
* Source/bmalloc/bmalloc/DebugHeap.cpp: Ditto.
  (bmalloc::DebugHeap::memalign): Ditto.

Canonical link: https://commits.webkit.org/279189@main
